TERMS.md

```diff
@@ -0,0 +1,97 @@
+	- A Service name is not in itself a Namespace, but its contents must be defined within *at least one* **Namespace**.  The Namespace in question is referenced or defined from within the Service definition.
+	- A Service must have its fundamental definition in **one** single .yaml formatted file, although this file might use references to other parts.  In other words, all Interfaces that make up a service must be mentioned within this file*.
+	- A file can contain a maximum of one Service.
+
+_>	(To be further determined - how complete is our include/reference concept?)_
+
+- **Namespace**
+	- A namespace is a way to separate definitions in named groups.
+	- Namespaces are used to ensure that local names will not clash with identically named items in other namespace.
+	- Namespaces usually also separate what objects can be *seen* or *reached* from within a part of a program.
+	- Namespaces can contain other Namespaces, creating a hierarchy.
+	- How visibility/reachability is handled *might* be specific to the target language or environment, but VSC generally assumes that items within different (non-parent) namespaces are isolated from each other unless otherwise specified, and that common hierarchy rules apply (i.e. everything within a child namespace can see and reach items in any of its parent namespaces without a specific required statement of reference or inclusion).
+- **Interface**
+	- (General)
+		- very broad term...
+	- (VSC)
+	- A VSC **Interface** is *primarily* defined by a collection of **Methods**.
+	- An Interface name is not itself a Namespace but must be defined within a Namespace.
+	- All Methods defined or referred to by an Interface are required to be within the same Namespace, i.e. it is not possible to build an Interface by referring to Methods that have been defined in different Namespaces.
+	- The VSC interface definition language strives to usable in all areas of a computing system.   The VSC Interface object is therefore a pure software function focused definition, which does not define details around communication hardware, target language, communication protocols etc.  Such information is added through supplementary information later.
+- **Method**
+	- A method is a function with a (mandatory) name, and a set of **input parameters**, **output parameters**, and **error conditions**, all of which are optional and may be empty/undefined.
+	- Like most other objects, a Method is always defined in the context of a Namespace, but this is most often implicitly determined because the Method is defined within the context of an Interface/Service
+- **Error**
+	- A VSC Error definition is an error condition that can appear while (attempting to) execute a given Method.
```

Comment:
> Could be an idea to say something on what type of errors that are intended to be specified in VSC

I don't disagree with what you're saying but I just feel it's on a different track than where most of my thinking is right now.  What gets standardized as standard services, standard errors, and so on, is independent from the technical capability to do so.  I feel like we need to try to separate the technical specification of the "Interface Description Language" from any concerns of standardized definitions, which in my thinking is something that gets decided later.

From the TERMS:
"_A VSC Error definition is an error condition that can appear while (attempting to) execute a given Method_"

Here is also why I would prefer to give the language **its own name**.  In this text I wrote "VSC Error definition" simply because I have no better name than "VSC.   In this case I meant VSC just as a placeholder for "The IDL"  so I understand why it led you to think about the "VSC project" and what it will standardize or not.  We're struggling from not separating the different type of work here.

I wrote more about this in some texts I hope to share soon.

> If it only are "business logic errors" as seen by the service implementation (e.g. seat stuck, movement rejected for safety reasons) or also "infrastructure"-errors (e.g. no channel to server, no response from server, authentication error, protocol version mismatch, encoding/decoding error...)

Well if we are still talking about what the "VSC project" will standardize and not, I think it is jumping the gun to expect this to be decided now.  I mean, roughly speaking, _iff_ the project, meaning _its participants_, end up wanting to defining things like the transfer protocol, etc., then I think it is _possible_ that infrastructure errors become defined as a consequence of that.  Otherwise I expect they might now.  Or they might... because someone thinking about it realizes that some work needs to get done or it will difficult to actually use this stuff _in practice_.  Here, comparing the approach to gRPC is good because on the one hand they have defined some useful standard errors, but on the other it's messy (as you pointed out) because it mixes the two variations of errors together.

But you're on the same train of thought as I am.  Going back to the purely technical capability and best-practices, I already have considered and written down some "plans" for this.  Before I get that text out there:  briefly the idea is that the original interface-definition would normally only define the biz logic concerns & error conditions (in the way I propose in the text we discuss below), and that some _layer_ ought then to be used to apply protocol specific concerns.

Reply:
Re-reading the comment, I realize I was suffering from the problem of "VSC" meaning many things.  I thought you meant the VSC as a project, more than the technology (or for that matter, which part of the technology such as the "core" IDL, vs other layers.     I also did not quite pick up that you mentioned *type* of error.   If you meant to highlight the separation of business logic and protocol errors, then yes it will be covered.  That is part of the strategy of being able to define multiple independent error-returns from one , and through layering separate the two types of errors in definitions.